spec/spec_helper.rb

```diff
@@ -9,6 +9,17 @@
 require File.expand_path("../../config/environment", __FILE__)
 require 'application_helper'
 
+# Fixing a bug with miq-kernel, should be updated to match this or just removed
```

Comment:
I agree.  This should be fixed in the future.  I don't remember what doesn't work in irb but we should fix miq-kernel to remove the patch if required.  I don't know why we need it in irb vs. only in console, if the latter is fine, maybe this patch can be isolated to the `console do` block in the config/application.rb and not contaminate the aruba setup.

Reply:
Yeah, this was part of this PR that I figured would never actually make it into master, but put it here for now for a reminder that it is a PITA, and currently causing issues with other libs.

Since I didn't explain, basically the bug `miq-kernel` is that it will indiscriminately apply `dirname` of the `caller`, even if the path is an absolute path.  Part of aruba (for some reason) passes paths that actually don't need any "relativity", and possibly are absolute paths (I forget... it has been a bit..), so that fails with this monkey patch.

Anyway, I like your ideas, and when I get back to working on this, we should definitely go with one of those options.